dm/relay/binlog_writer.go

```diff
@@ -105,32 +176,46 @@ func (w *BinlogWriter) Close() error {
 
 	w.file = nil
 	w.offset.Store(0)
-	w.uuid = ""
-	w.filename = ""
+	w.uuid.Store("")
+	w.filename.Store("")
+	w.input = nil
 
+	if writeErr := w.Error(); writeErr != nil {
```

Comment:
return w.Error()

Reply:
If w.Error()!=nil, we should return the error of file close, L174